jetcd-core/src/test/java/io/etcd/jetcd/ClientBuilderTest.java

```diff
@@ -53,4 +58,45 @@ public void testMaxInboundMessageSize() throws URISyntaxException {
 
     assertThat(channelBuilder).hasFieldOrPropertyWithValue("maxInboundMessageSize", value);
   }
+
+  @Test
+  public void testNamespace() {
+    class TestCase {
+      public ByteSequence namespace;
+      public ByteSequence expectedNamespace;
+
+      public TestCase(ByteSequence namespace, ByteSequence expectedNamespace) {
+        this.namespace = namespace;
+        this.expectedNamespace = expectedNamespace;
+      }
+    }
+    List<TestCase> testCases = Arrays.asList(
+        // namespace setting and expected namespace
+        new TestCase(ByteSequence.EMPTY, ByteSequence.EMPTY),
+        new TestCase(ByteSequence.from("/", Charsets.UTF_8), ByteSequence.EMPTY),
+        new TestCase(ByteSequence.from("/namespace1", Charsets.UTF_8),
+            ByteSequence.from("/namespace1/", Charsets.UTF_8)),
+        new TestCase(ByteSequence.from("/namespace1/", Charsets.UTF_8),
+            ByteSequence.from("/namespace1/", Charsets.UTF_8)),
+        new TestCase(ByteSequence.from("namespace2/", Charsets.UTF_8),
+            ByteSequence.from("namespace2/", Charsets.UTF_8))
+    );
+    testCases.forEach(testCase -> {
+      try {
+        final ClientBuilder builder =  Client.builder().endpoints(new URI("http://127.0.0.1:2379"))
+            .namespace(testCase.namespace);
+        final ClientConnectionManager connectionManager = new ClientConnectionManager(builder);
+        assertEquals(testCase.expectedNamespace, connectionManager.getNamespace());
```

Comment:
could you use `assertThat(actual).isEqualTo(expected)` from assertJ instead of assertEquals? I think it is best to have a consistent testing tool though out the codebase.

Reply:
done